packages/wrangler/src/__tests__/secret.test.ts

```diff
@@ -44,83 +48,180 @@ describe("wrangler secret", () => {
       });
 
-      mockPutRequest(
-        { name: "the-secret-name", text: "the-secret" },
-        "some-env",
-        false
-      );
-      await runWrangler(
-        "secret put the-key --name script-name --env some-env --legacy-env false"
-      );
+      it("should error without a script name", async () => {
+        let error: Error | undefined;
+        try {
+          await runWrangler("secret put the-key");
+        } catch (e) {
+          error = e as Error;
+        }
+        expect(std.out).toMatchInlineSnapshot(`""`);
+        expect(std.err).toMatchInlineSnapshot(`
+          "Missing script name
 
-      expect(std.out).toMatchInlineSnapshot(`
-        "🌀 Creating the secret for script script-name (some-env)
-        ✨ Success! Uploaded secret the-key"
-      `);
-      expect(std.err).toMatchInlineSnapshot(`""`);
+          [32m%s[0m If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new."
+        `);
+        expect(error).toMatchInlineSnapshot(`[Error: Missing script name]`);
+      });
     });
 
-    it("should error without a script name", async () => {
-      let error: Error | undefined;
-      try {
-        await runWrangler("secret put the-key");
-      } catch (e) {
-        error = e as Error;
-      }
-      expect(std.out).toMatchInlineSnapshot(`""`);
-      expect(std.err).toMatchInlineSnapshot(`
-        "Missing script name
+    describe("non-interactive", () => {
+      const mockStdIn = useMockStdin({ isTTY: false });
 
-        [32m%s[0m If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new."
-      `);
-      expect(error).toMatchInlineSnapshot(`[Error: Missing script name]`);
+      it("should create a secret, from piped input", async () => {
+        mockPutRequest({ name: "the-key", text: "the-secret" });
+        // Pipe the secret in as three chunks to test that we reconstitute it correctly.
+        mockStdIn.send("the", "-", "secret");
+        await runWrangler("secret put the-key --name script-name");
+
+        expect(std.out).toMatchInlineSnapshot(`
+          "🌀 Creating the secret for script script-name
+          ✨ Success! Uploaded secret the-key"
+        `);
+        expect(std.warn).toMatchInlineSnapshot(`""`);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+      });
+
+      it("should error if the piped input fails", async () => {
+        mockPutRequest({ name: "the-key", text: "the-secret" });
+        mockStdIn.throwError(new Error("Error in stdin stream"));
+        await expect(
+          runWrangler("secret put the-key --name script-name")
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`"Error in stdin stream"`);
+
+        expect(std.out).toMatchInlineSnapshot(`""`);
+        expect(std.warn).toMatchInlineSnapshot(`""`);
+      });
+
+      describe("with accountId", () => {
+        mockAccountId({ accountId: "" });
```

Comment:
oh huh, do we accept a blank string as an invalid accountid? should we?

Reply:
Yeah, I was wondering about this... At the moment we just do tests like `if (accountId)`, which is why we can get away with setting `accountId` to the empty string. But if we tightened that up then we would need to make this better.

Since you brought it up, I'll improve that now.